### PR TITLE
Add docs on using Bionic with NativeAOT

### LINKED ISSUE #87340: Unable to publish NativeAOT project for linux-bionic-arm64
### Description

Building a simple Hello World sample works for the linux-bionic-x64 and linux-bionic-arm64 rids, but publishing AOT fails for linux-bionic-arm64, but works for linux-bionic-x64. The following error is thrown immediately when attempting the build,

```
/home/emmaus/.nuget/packages/microsoft.dotnet.ilcompiler/8.0.0-preview.6.23307.4/build/Microsoft.NETCore.Native.Publish.targets(75,5): error : The PrivateSdkAssemblies ItemGroup is required for _ComputeAssembliesToCompileToNative [/home/emmaus/test/test.csproj]
```

### Reproduction Steps

A simple project with the required nuget packages already referenced here, https://github.com/emmauss/test-aot.
Restore and build it with 
`dotnet publish -r linux-bionic-arm64 -c Debug -p:SysRoot=/.tools/android-rootfs/android-ndk-r21/sysroot`. The error is immediately thrown.

### Expected behavior

The project quickly builds and a compatible executable is generated.

### Actual behavior

The following error is displayed,
```
/home/emmaus/.nuget/packages/microsoft.dotnet.ilcompiler/8.0.0-preview.6.23307.4/build/Microsoft.NETCore.Native.Publish.targets(75,5): error : The PrivateSdkAssemblies ItemGroup is required for _ComputeAssembliesToCompileToNative [/home/emmaus/test/test.csproj]
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

OS: Fedora 38 on WSL2 running on Windows 11 x64 host
.Net Version: 8.0.0-preview.6.23307.4 daily build

### Other information

_No response_

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Resolves #87340.

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/coreclr/nativeaot/docs/android-bionic.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+* Update your PATH to include `{NDK_ROOT}\toolchains\llvm\prebuilt\{OS_ARCH}\bin`. Replace `{NDK_ROOT}` with the path where you extracted the NDK, and replace `{OS_ARCH}` with your host OS and architecture (e.g. windows-x86_64 or linux-x86_64). Make sure this entry is first in your path - we need to make sure that running `clang` will execute `clang` from this directory.
+* You can either create a new project or use an existing project. For this guide, let's create a new one:
+  ```sh
+  $ dotnet new console -o HelloBionic --aot
+  $ cd HelloBionic
+  ```
+  Note: the `--aot` parameter added `<PublishAot>true</PublishAot>` to the new project.
+* Publish the project:
+  ```sh
+  $ dotnet publish -r linux-bionic-arm64 -p:DisableUnsupportedError=true -p:PublishAotUsingRuntimePack=true
```
Comment (vitek-karas) [inline src/coreclr/nativeaot/docs/android-bionic.md:19]: Are the two properties going away eventually? If not, we should probably at least mention them in this doc. Their names don't make it super clear what they're doing.

File: src/coreclr/nativeaot/docs/android-bionic.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ```
+* You should have a binary under `bin\Release\net8.0\linux-bionic-arm64\publish`. Copy this to an Android device. Either `adb push` or using some GUI.
```
Comment (jkotas) [inline src/coreclr/nativeaot/docs/android-bionic.md:21]:
```suggestion
* You should have a binary under `bin\Release\net8.0\linux-bionic-arm64\publish`. Copy it to an Android device. Either `adb push` or using some GUI.
```

File: src/coreclr/nativeaot/docs/android-bionic.md (hunk 1 above)
Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/docs/android-bionic.md:19]: The `PublishAotUsingRuntimePack` is supposed to go away. We currently have to set this for iDevices for the same reason. We'd like to use the runtime pack for all configurations, not just iDevices/Bionic. There's an 8.0 issue on it.

File: src/coreclr/nativeaot/docs/android-bionic.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Using Native AOT with Android Bionic
+
+Starting with .NET 8 Preview 7, it's possible to build shared libraries and command line executables for Android.
+
+Not a full Android experience is available - it's only possible to publish for two Bionic RID: bionic-arm64 and bionic-x64. Publishing for Android RIDs (android-arm64/android-x64) is not possible. This limited experience corresponds to building with [Android NDK](https://developer.android.com/ndk) from Native code - the limitations are similar. Interop with Java needs to be done manually through JNI, if necessary.
```
[the remark this answers is not in the document]
Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/docs/android-bionic.md:5]:
```suggestion
Not a full Android experience is available - it's only possible to publish for two Bionic RID: linux-bionic-arm64 and linux-bionic-x64. Publishing for Android RIDs (android-arm64/android-x64) is not possible. This limited experience corresponds to building with [Android NDK](https://developer.android.com/ndk) from Native code - the limitations are similar. Interop with Java needs to be done manually through JNI, if necessary.

The minimum API level is 21 at the time of writing the document, but search for AndroidApiLevelMin in this repo for more up-to-date information.
```

**Review by jkotas — APPROVED** [review]
LGTM

**Pull request merged**